### ISSUE TITLE
💅 useReadonlyClass disregards increment operator when used in template literals or function arguments

### ISSUE THREAD

**Issue by vladkrasn** [biomejs/biome#6775]
### Environment information

```bash
Biome playground
```

### Rule name

useReadonlyClass

### Playground link

https://biomejs.dev/playground/?lintRules=useReadonlyClassProperties&code=YwBsAGEAcwBzACAAQwBvAHUAbgB0AGUAcgAgAHsACgAgACAAcAByAGkAdgBhAHQAZQAgAGMAbwB1AG4AdABlAHIAOgAgAG4AdQBtAGIAZQByAAoAIAAgAGMAbwB1AG4AdAAoACkAIAB7AAoAIAAgACAAIABjAG8AbgBzAG8AbABlAC4AbABvAGcAKAB0AGgAaQBzAC4AYwBvAHUAbgB0AGUAcgArACsAKQAKACAAIAAgACAAYwBvAG4AcwB0ACAAYwBvAHUAbgB0AGUAcgBTAHQAcgBpAG4AZwAgAD0AIABgACQAewB0AGgAaQBzAC4AYwBvAHUAbgB0AGUAcgArACsAfQBgAAoAIAAgAH0ACgB9AA%3D%3D

### Expected result

++ mutates the argument, so it shouldn't be readonly

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct
